infrastructures/infrastructure-azure/src/test/java/org/ow2/proactive/resourcemanager/nodesource/infrastructure/AzureInfrastructureTest.java

```diff
@@ -0,0 +1,353 @@
+
+        when(node.getProperty(AzureInfrastructure.INSTANCE_ID_NODE_PROPERTY)).thenReturn("123");
+
+        when(node.getNodeInformation()).thenReturn(nodeInformation);
+
+        when(node.getProActiveRuntime()).thenReturn(proActiveRuntime);
+
+        when(nodeInformation.getName()).thenReturn("nodename");
+
+        azureInfrastructure.nodesPerInstances.put("123", Sets.newHashSet("nodename"));
+
+        azureInfrastructure.removeNode(node);
+
+        verify(proActiveRuntime).killNode("nodename");
+
+        verify(connectorIaasController).terminateInstance("node_source_name", "123");
+
+        assertThat(azureInfrastructure.nodesPerInstances.isEmpty(), is(true));
+
+    }
+
+    @Test
+    public void testNotifyAcquiredNode() throws ProActiveException, RMException {
+
+        when(nodeSource.getName()).thenReturn("Node source Name");
+        azureInfrastructure.nodeSource = nodeSource;
+        azureInfrastructure.configure("clientId",
+                                      "secret",
+                                      "domain",
+                                      "subscriptionId",
+                                      "authenticationEndpoint",
+                                      "managementEndpoint",
+                                      "resourceManagerEndpoint",
+                                      "graphEndpoint",
+                                      "test.activeeon.com",
+                                      "http://localhost:8088/connector-iaas",
+                                      "image",
+                                      "Standard_D1_v2",
+                                      "vmUsername",
+                                      "vmPassword",
+                                      "vmPublicKey",
+                                      "resourceGroup",
+                                      "region",
+                                      "2",
+                                      "3",
+                                      "wget -nv test.activeeon.com/rest/node.jar",
+                                      "192.168.1.0/24",
+                                      true,
+                                      "-Dnew=value");
+
+        azureInfrastructure.connectorIaasController = connectorIaasController;
+
+        when(node.getProperty(AzureInfrastructure.INSTANCE_ID_NODE_PROPERTY)).thenReturn("123");
+
+        when(node.getNodeInformation()).thenReturn(nodeInformation);
+
+        when(nodeInformation.getName()).thenReturn("nodename");
+
+        azureInfrastructure.notifyAcquiredNode(node);
+
+        assertThat(azureInfrastructure.nodesPerInstances.get("123").isEmpty(), is(false));
+        assertThat(azureInfrastructure.nodesPerInstances.get("123").size(), is(1));
+        assertThat(azureInfrastructure.nodesPerInstances.get("123").contains("nodename"), is(true));
+
+    }
+
+    @Test
+    public void testGetDescription() {
```

Comment:
More verbose name. Your tests, test assumptions. Put that assumption in the test name. Those are like your specs.